hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/services/AbfsRestOperation.java

```diff
@@ -278,14 +291,33 @@ public void signRequest(final AbfsHttpOperation httpOperation,
    */
   private boolean executeHttpOperation(final int retryCount,
     TracingContext tracingContext) throws AzureBlobFileSystemException {
-    AbfsHttpOperation httpOperation = null;
+    AbfsHttpOperation httpOperation;
     try {
       // initialize the HTTP request and open the connection
-      httpOperation = createNewHttpOperation();
+      httpOperation = createHttpOperation();
       incrementCounter(AbfsStatistic.CONNECTIONS_MADE, 1);
       tracingContext.constructHeader(httpOperation);
 
-      signRequest(httpOperation, hasRequestBody ? bufferLength : 0);
+      switch(client.getAuthType()) {
```

Comment:
why are we pulling back code of signRequest to the executeHttpOperation

Reply:
Refactored